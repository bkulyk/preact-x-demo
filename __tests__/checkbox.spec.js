import React from 'react';
import Checkbox from '@bitchin/checkbox';
import { shallow } from './enzyme';

it('render correctly with defaults', () => {
  expect(shallow(<Checkbox />)).toMatchSnapshot();
});

it('render correctly with indeterminate', () => {
  expect(shallow((
    <Checkbox
      label="on/off"
      indeterminate
      className="ind-class"
      id="ind-checkbox"
    />
  ))).toMatchSnapshot();
});

it('render correctly with some props', () => {
  expect(shallow((
    <Checkbox
      label="on/off"
      disabled
      className="extra-class"
      id="my-checkbox"
    />
  ))).toMatchSnapshot();
});
