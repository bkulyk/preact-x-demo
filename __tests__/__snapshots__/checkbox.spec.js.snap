// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with defaults 1`] = `
<div
  className="mdc-form-field"
>
  <div
    className="mdc-checkbox"
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={false}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
</div>
`;

exports[`render correctly with indeterminate 1`] = `
<div
  className="mdc-form-field ind-class"
>
  <div
    className="mdc-checkbox"
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={false}
      id="ind-checkbox"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    htmlFor="ind-checkbox"
  >
    on/off
  </label>
</div>
`;

exports[`render correctly with some props 1`] = `
<div
  className="mdc-form-field extra-class"
>
  <div
    className="mdc-checkbox mdc-checkbox--disabled"
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={true}
      id="my-checkbox"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    htmlFor="my-checkbox"
  >
    on/off
  </label>
</div>
`;
